File: lifeos-shared/src/storage/syncService.ts
import { PowerSyncDatabase } from '@powersync/react-native';
import { SCHEMA } from '../../../innerscape-soma/lib/sync/schema'; // Reference the schema

/**
 * @fileoverview Shared PowerSync Service
 * @module storage/syncService
 * 
 * APEX Contract:
 * - Provides unified sync logic for Turso/PowerSync
 */

export const createSyncService = (db: PowerSyncDatabase) => {
  return {
    async connect(token: string) {
      console.log('[APEX] Connecting to PowerSync');
      try {
        await (db as any).connect({
          fetchToken: async () => token,
          endpoint: 'https://powersync.innerscape.app'
        });
      } catch (err) {
        console.error('[APEX] PowerSync connection failed', err);
      }
    },

    async disconnect() {
      await db.disconnect();
    },

    // Helper to log activities cross-app
    async recordActivity(activity: {
      userId: string;
      action: string;
      entityType: string;
      entityId: string;
      content?: string;
    }) {
      await db.execute(
        'INSERT INTO activities (id, user_id, action, entity_type, entity_id, content, timestamp) VALUES (?, ?, ?, ?, ?, ?, ?)',
        [
          crypto.randomUUID(),
          activity.userId,
          activity.action,
          activity.entityType,
          activity.entityId,
          activity.content || null,
          Date.now()
        ]
      );
    }
  };
};

File: _archived/Feelings APP/lib/domain/services/sync_service.dart
// lib/domain/services/sync_service.dart
import 'package:powersync/powersync.dart';

/**
 * APEX Contract: Soma Sync Service
 * Purpose: Local-first SQLite sync with Turso cloud via PowerSync
 */

class SyncService {
  late PowerSyncDatabase db;

  Future<void> initialize(String userId) async {
    print('[APEX] Initializing PowerSync for user $userId');
    
    // Schema matches lifeos-backend/src/db/schema.ts
    final schema = Schema([
      Table('emotional_context', [
        Column.text('id'),
        Column.text('user_id'),
        Column.integer('energy'),
        Column.integer('valence'),
        Column.text('dominant_feeling'),
        Column.text('body_sensation'),
        Column.integer('timestamp'),
      ]),
    ]);

    db = PowerSyncDatabase(schema: schema, path: 'innerscape_soma.db');
    await db.initialize();
    
    // Connection credentials would be provided by backend/clerk
    /*
    await db.connect(
      connector: PowerSyncConnector(
        endpoint: 'https://innerscape.powersync.com',
        userId: userId,
      ),
    );
    */
  }

  Future<void> saveCheckIn({
    required int energy,
    required int valence,
    required String dominantFeeling,
    String? bodySensation,
  }) async {
    final timestamp = DateTime.now().millisecondsSinceEpoch;
    final id = 'checkin_${timestamp}';

    await db.execute(
      '''INSERT INTO emotional_context (id, user_id, energy, valence, dominant_feeling, body_sensation, timestamp) 
         VALUES (?, ?, ?, ?, ?, ?, ?)''',
      [id, 'test-user-123', energy, valence, dominantFeeling, bodySensation, timestamp],
    );
    
    print('[APEX] Check-in saved locally and queued for sync');
  }
}
